keps/sig-auth/4317-pod-certificates/README.md

```diff
@@ -0,0 +1,1428 @@
+* pkg/controller/certificates/podcertificaterequest_controller_test.go
+* pkg/controller/certificates/signer/signer_test.go
+* pkg/kubelet/podcertificate/podcertificate_manager_test.go
+* pkg/registry/certificates/podcertificaterequest/storage/storage_test.go
+* pkg/registry/certificates/podcertificaterequest/strategy_test.go
+* pkg/volume/projected/projected_test.go
+* plugin/pkg/admission/noderestriction/admission_test.go
+* plugin/pkg/auth/authorizer/rbac/bootstrappolicy/controller_policy_test.go
+* plugin/pkg/auth/authorizer/rbac/bootstrappolicy/policy_test.go
+* staging/src/k8s.io/apiserver/pkg/authentication/request/x509/x509_test.go
+* staging/src/k8s.io/client-go/rest/config_test.go
+* staging/src/k8s.io/component-helpers/kubernetesx509/kubernetesx509_test.go
+
+<!--
+Additionally, for Alpha try to enumerate the core package you will be touching
+to implement this enhancement and provide the current unit coverage for those
+in the form of:
+- <package>: <date> - <current test coverage>
+The data can be easily read from:
+https://testgrid.k8s.io/sig-testing-canaries#ci-kubernetes-coverage-unit
+
+This can inform certain test coverage improvements that we want to do before
+extending the production code to implement this enhancement.
+-->
+
+- `<package>`: `<date>` - `<test coverage>`
+
+##### Integration tests
+
+<!--
+Integration tests are contained in k8s.io/kubernetes/test/integration.
+Integration tests allow control of the configuration parameters used to start the binaries under test.
+This is different from e2e tests which do not allow configuration of parameters.
+Doing this allows testing non-default options and multiple different and potentially conflicting command line options.
+-->
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+-->
+
+- <test>: <link to test coverage>
+
+##### e2e tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+
+We expect no non-infra related flakes in the last month as a GA graduation criteria.
+-->
+
+* test/e2e/auth/projected_podcertificate.go: Checks end-to-end issuance and
+  usage of pod certificates issued by `kubernetes.io/kube-apiserver-client-pod`.
+  The test will create an agnhost pod with a mounted credential bundle, and
+  agnhost will use the credential bundle to connect to kube-apiserver.
+
+### Graduation Criteria
+
+For Alpha:
+* The feature will be gated behind feature flags and alpha API
+enablement.
+* An e2e test will exercise the feature in order to prevent
```

Comment:
We don't enable alpha feature gates in our regular e2es, so make sure to add them to appropriate test suite which does it.

Reply:
Will do.  We had to do the same thing for ClusterTrustBundles.